src/libpmemfile-core/dir.c

```diff
@@ -1215,14 +1215,24 @@ pmemfile_rmdir(PMEMfilepool *pfp, const char *path)
 }
 
 static int
-_pmemfile_chdir(PMEMfilepool *pfp, struct pmemfile_vinode *dir)
+_pmemfile_chdir(PMEMfilepool *pfp, struct pmemfile_cred *cred,
+		struct pmemfile_vinode *dir)
 {
-	if (!vinode_is_dir(dir)) {
+	struct inode_perms dir_perms;
+	vinode_get_perms(dir, &dir_perms);
+
+	if (!PMEMFILE_S_ISDIR(dir_perms.flags)) {
 		vinode_unref_tx(pfp, dir);
 		errno = ENOTDIR;
 		return -1;
 	}
 
+	if (!can_access(cred, &dir_perms, PFILE_WANT_EXECUTE)) {
```

Comment:
BTW, what happens, if the user doesn't have permission to chdir to the root directory? Can the user open the pmemfile pool?